weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ObjectsPage/ObjectDeleteButtons.tsx

```diff
@@ -13,13 +18,32 @@ export const DeleteObjectButtonWithModal: React.FC<{
 }> = ({objVersionSchema, overrideDisplayStr}) => {
   const {useObjectDeleteFunc} = useWFHooks();
   const closePeek = useClosePeek();
+  const {isPeeking} = useContext(WeaveflowPeekContext);
+  const routerContext = useWeaveflowCurrentRouteContext();
+  const history = useHistory();
   const {objectVersionsDelete} = useObjectDeleteFunc();
   const [deleteModalOpen, setDeleteModalOpen] = useState(false);
 
   const deleteStr =
     overrideDisplayStr ??
     `${objVersionSchema.objectId}:v${objVersionSchema.versionIndex}`;
 
+  const onSuccess = () => {
+    if (isPeeking) {
+      closePeek();
+    } else {
+      history.push(
+        routerContext.objectVersionsUIUrl(
+          objVersionSchema.entity,
+          objVersionSchema.project,
+          {
+            objectName: objVersionSchema.objectId,
```

Comment:
objectName vs. objectId is pretty confusing

Reply:
yeah, i'll add a ticket to the backlog